src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandler.java

```diff
@@ -323,4 +332,48 @@ protected Mono<AuthorizationOutput> redirectionAuthRequestPipeline(
 
         return redirectionAuthRequestPipeline(authorizationData, touchpoint);
     }
+
+  /**
+   * Emit Wallet Used event on wallet queue. The semantic
+   * of this method is fire-and-forget, so any action performed by this
+   * method is executed asynchronously.
+   * e.g. doOnNext(_ -> fireWalletLastUsageEvent(...))
+   */
+    private void fireWalletLastUsageEvent(
+            TransactionRequestAuthorizationCommand command,
+            TransactionActivated transactionActivated
+    ) {
+      final var wallet = switch (command.getData().authDetails()) {
+        case WalletAuthRequestDetailsDto walletData -> Mono.just(walletData);
+        default -> Mono.<WalletAuthRequestDetailsDto>empty();
+      };
+
+      walletAsyncQueueClient.ifPresent(
+              queueClient -> wallet.flatMap(walletData -> tracingUtils.traceMono(
+                              this.getClass().getSimpleName(),
+                              (tracingInfo) -> queueClient.fireWalletLastUsageEvent(
+                                      walletData.getWalletId(),
+                                      transactionActivated.getClientId(),
+                                      tracingInfo
+                              )
+                      ).doOnError(
+                              exception -> log.error(
+                                      "Failed to send event WALLET_USED for transactionId {}, wallet {}, clientId {} - error {}",
+                                      transactionActivated.getTransactionId(),
+                                      walletData.getWalletId(),
+                                      transactionActivated.getClientId(),
+                                      exception.getMessage()
+                              )
+                      )
```

Comment:
```suggestion
                      ).doOnError(
                                        exception -> log.error(
                                                "Failed to send event WALLET_USED for transactionId: [%s], wallet: [%s], clientId: [%s]"
                                                        .formatted(
                                                                transactionActivated.getTransactionId(),
                                                                walletData.getWalletId(),
                                                                transactionActivated.getClientId()
                                                        ),
                                                exception
                                        )
                                )
```
please use the logger.error(message, throwable) method so that stacktrace is add to log